### PR TITLE
Feature/#217 create tree component

### LINKED ISSUE #217: Create a tree component
this one is complex :)

Sintax could be something like:

```
[Folder] Name of the folder
   [Folder] Subfolder
       [File]  Name of the file
[Folder] Another folder name
```

And aspect

Image

### PR DESCRIPTION
**Pull request by gustedeveloper**
Close #217

### CODE REVIEW

File: src/common/components/mock-components/front-rich-components/file-tree/file-tree.business.ts [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    calculatedSize.height,
+    currentSize.height,
+    id,
+    coords.x,
+    coords.y,
+    updateShapeSizeAndPosition,
+  ]);
+};
+
+export const useFileTreeResizeOnSizeChange = (
```
Comment (brauliodiez) [inline src/common/components/mock-components/front-rich-components/file-tree/file-tree.business.ts:187]: Maybe you can keep this two hooks private and create a third hook that wraps this two, something like

useFilteTreeResize

File: src/common/components/mock-components/front-rich-components/file-tree/file-tree.business.ts [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+}
+
+interface FileTreeDynamicSizeParams {
+  width: number;
+  elementHeight: number;
+  paddingY: number;
+  baseRestrictions: ShapeSizeRestrictions;
+}
+
+export const parseFileTreeText = (text: string): FileTreeItem[] => {
```
Comment (brauliodiez) [inline src/common/components/mock-components/front-rich-components/file-tree/file-tree.business.ts:77]: It would be a good idea to add unit tests to this method, including corner cases

File: src/common/components/mock-components/front-rich-components/file-tree/file-tree.business.ts (hunk 1 above)
Comment (gustedeveloper) [reply src/common/components/mock-components/front-rich-components/file-tree/file-tree.business.ts]: Both private hooks wrapped into useFileTreeResize hook

File: src/pods/properties/components/select-size/select-size-v2/index.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export * from './select-size-v2.component';
```
[the remark this answers is not in the document]
Comment (gustedeveloper) [reply src/pods/properties/components/select-size/select-size-v2/index.ts:1]: I've created the issue https://github.com/Lemoncode/quickmock/issues/791 for the future replacement of select size

File: src/common/components/mock-components/front-rich-components/file-tree/file-tree.business.ts (hunk 2 above)
Comment (gustedeveloper) [reply src/common/components/mock-components/front-rich-components/file-tree/file-tree.business.ts:78]: Unit tests done!
